### PR TITLE
Feature/multi audio

### PR DESCRIPTION
**Pull request by gfronza**

### CODE REVIEW

File: examples/encoding/create_simple_encoding_hls_with_multiple_audio_groups_and_stream_conditions/main.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+	audioMuxing128kStream := models.StreamItem{
+		StreamID: aacStream128kResp.Data.Result.ID,
+	}
+	audioMuxing98kStream := models.StreamItem{
+		StreamID: aacStream98kResp.Data.Result.ID,
+	}
+
+	videoMuxing1080pOutput := models.Output{
+		OutputID:   s3OutputResp.Data.Result.ID,
+		OutputPath: stringToPtr("golang_test/video/1080p"),
```
Comment (moser4035) [inline examples/encoding/create_simple_encoding_hls_with_multiple_audio_groups_and_stream_conditions/main.go]: Maybe define the baseOutputPath as a variable at the beginning and only change the relative paths here
e.g.:
```
	outputBasePath := fmt.Sprintf("your/output/base")
```
```
	videoMuxing1080pOutput := models.Output{
		OutputID:   s3OutputResp.Data.Result.ID,
		OutputPath: stringToPtr(fmt.Sprintf("%s/video/1080p", outputBasePath)),
		ACL:        acl,
	}
```

File: examples/encoding/create_simple_encoding_hls_with_multiple_audio_groups_and_stream_conditions/main.go [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+		status = *statusResp.Data.Result.Status
+		if status == "ERROR" {
+			fmt.Println("error in Encoding Status")
+			fmt.Printf("%+v\n", statusResp)
+			return
+		}
+	}
+
+	// Delete Encoding
+	deleteResp, err := encodingS.Delete(*encodingResp.Data.Result.ID)
```
Comment (moser4035) [inline examples/encoding/create_simple_encoding_hls_with_multiple_audio_groups_and_stream_conditions/main.go]: Hmm, I think this shouldn't be done here in this example.

File: examples/encoding/create_simple_encoding_hls_with_multiple_audio_groups_and_stream_conditions/main.go [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+	for status != "FINISHED" {
+		time.Sleep(10 * time.Second)
+		statusResp, err := encodingS.RetrieveStatus(*encodingResp.Data.Result.ID)
+		if err != nil {
+			fmt.Println("error in Encoding Status")
+			fmt.Println(err)
+			return
+		}
+		// Polling and Printing out the response
+		fmt.Printf("%+v\n", statusResp)
```
Comment (moser4035) [inline examples/encoding/create_simple_encoding_hls_with_multiple_audio_groups_and_stream_conditions/main.go]: Maybe find a way to print better information here. This doesn't really show useful information. Is there a way to pretty print the object?

Comment (moser4035) [reply examples/encoding/create_simple_encoding_hls_with_multiple_audio_groups_and_stream_conditions/main.go]: You can use a helper method like [here](https://github.com/bitmovin/bitmovin-go/blob/master/examples/infrastructure/create_aws_infrastructure.go#L78)

**Comment by gfronza** [comment]
@moser4035 thanks for the ideas! Changes applied.

File: models/hls.go [hunk 4]
```diff
@@ -38,20 +38,35 @@ type LiveHLSManifest struct {
 	TimeShift  *float64 `json:"timeShift,omitempty"`
 }
 
+type HLSAudioGroupDefinition struct {
+	Name     string `json:"name,omitempty"`
+	Priority int    `json:"priority,omitempty"`
```
Comment (moser4035) [inline models/hls.go]: If this is optional i think this should be a `*int`

File: models/hls.go [hunk 5]
```diff
@@ -38,20 +38,35 @@ type LiveHLSManifest struct {
 	TimeShift  *float64 `json:"timeShift,omitempty"`
 }
 
+type HLSAudioGroupDefinition struct {
+	Name     string `json:"name,omitempty"`
```
Comment (moser4035) [inline models/hls.go]: If this is optional I think this should be a `*string`

Comment (gfronza) [reply models/hls.go]: `Name` is mandatory.

**Pull request merged**